### PR TITLE
Remove vending useAsync calls and remove it all together

### PR DESCRIPTION
**Pull request by razzeee**

### CODE REVIEW

File: backend/app/vending/__init__.py
Comment (kymckay) [inline backend/app/vending/__init__.py:229]: GitHub prevents me commenting on unchanged lines, but on L207 of this file, the return type should match what is declared in the function header. It's currently causing issues with vending link retrieval in the frontend (a query condition falsely becomes true and tries to proceed to dashboard link retrieval).

**Review by kymckay — CHANGES_REQUESTED** [review]
Some routes are giving an `invalid-appid` error:
- `/vendingapp/org.flatpak.qtdemo/setup`
- `/vendingapp/org.flatpak.qtdemo/tokens`

File: backend/app/vending/__init__.py
Comment (kymckay) [reply backend/app/vending/__init__.py:229]: I fixed in testing locally by just returning a default `VendingStatus` with `status="none"` or similar and all `False` values.

Was unable to push to your fork branch.

File: frontend/src/components/user/VendingLink.tsx
Comment (kymckay) [inline frontend/src/components/user/VendingLink.tsx:72]:
```suggestion
        {t((statusQuery.error || dashboardQuery.error) as string)}
```

nit

File: frontend/src/components/user/VendingLink.tsx
Comment (kymckay) [inline frontend/src/components/user/VendingLink.tsx:37]: This is the condition which is causing errors in the frontend (becomes true when it shouldn't), which is interesting because it doesn't happen on `main` even though the logic looks identical apart from the extra `.data` attribute in the chain.

**Review by kymckay — COMMENTED** [review]
Further common on my initial review around the `invalid-appid`: the errors don't occur on `main` and the settings pages behave as expected

File: backend/app/vending/__init__.py
Comment (razzeee) [reply backend/app/vending/__init__.py:234]: I really don't like that status is just a string :/

Comment (razzeee) [reply backend/app/vending/__init__.py:234]: But changed it

File: frontend/src/components/user/VendingLink.tsx
Comment (razzeee) [reply frontend/src/components/user/VendingLink.tsx:37]: Well I guess that was caused by the 201 and us then trying to access `details_submitted`?

File: frontend/src/components/application/AppVendingControls/SetupControls.tsx
Comment (kymckay) [inline frontend/src/components/application/AppVendingControls/SetupControls.tsx:42]: Credentials here is the last thing needed and then this LGTM

**Review by kymckay — APPROVED** [review]
Approved to unblock merge, one last change needed

File: frontend/src/components/application/AppVendingControls/SetupControls.tsx
Comment (razzeee) [reply frontend/src/components/application/AppVendingControls/SetupControls.tsx]: Should be fixed now

**Pull request merged**
